libs/response-extensions.js

```diff
@@ -79,6 +79,10 @@ module.exports.send = (options, req, res) => {
             data.on('end', cb)
 
             return
+          } else if (Promise.resolve(data) === data) { // http://www.ecma-international.org/ecma-262/6.0/#sec-promise.resolve
+            return data
+              .then(resolved => send(resolved, code, headers, cb))
+              .catch(err => send(err, code, headers, cb))
```

Comment:
Since headers were already set by this point, it should be possible to pass these as null this time around to avoid looping over them again correct?
```suggestion
              .then(resolved => send(resolved, code, null, cb))
              .catch(err => send(err, code, null, cb))
```